[Messenger] Added SQS Messenger docs

Nyholm opened the pull request:
This will close #13328 and fix #13093

Simperfit committed:
feat: add amazon sqs to docs

abdel-ben committed:
Update user_provider.rst

Fix provider name `backend_users` instead of `backend`

wouterj committed:
minor #14339 Update user_provider.rst (abdel-ben)

This PR was submitted for the 5.1 branch but it was merged into the 4.4 branch instead.

Discussion
----------

Update user_provider.rst

Fix provider name `backend_users` instead of `backend`

Commits
-------

34bacb548 Update user_provider.rst

Nyholm committed:
Added SQS Messenger docs

Nyholm commented:
This PR should target 5.1

Nyholm committed:
cs

Nyholm committed:
syntax

Nyholm committed:
Syntax fix

Nyholm commented:
There you go @OskarStark. Im done with this PR now

jderusse commented on messenger.rst:
```diff
@@ -1268,6 +1268,72 @@ during a request::
+    The transport will automatically create queues that are needed. This
+    can be disabled setting the "auto_setup" option to ``false``.
+
+A numbeXr of options can be configured via the DSN or via the ``options`` key
```
`number`

Nyholm replied:
Ah.. I was testing if the docker build script automatically updated.. :)

jderusse reviewed (approved):
Thank you for taking care of this PR @Nyholm

Nyholm committed:
Typo

jderusse commented:
should we talk about async-aws, and authentication mechanism that fallback to .ini files, EC2/Lambda roles, Eks webidentity... and key/secret are not required

Nyholm committed:
minor

Nyholm commented:
Hm.. Im not sure we should. Access key/secret is a really good way for new users. Advanced users will look at the code and find AsyncAws and find their way...  
We dont want to duplicate the docs from AsyncAws

javiereguiluz commented on messenger.rst:
```diff
@@ -1268,6 +1268,72 @@ during a request::
+
+.. versionadded:: 5.1
+
+    The Amazon SQS transport has been added in Symfony 5.1
```
We'll make this change while merging, but for your information, the `versionadded` messages are standardized as: `XXX was introduced in Symfony N.N.`

Nyholm replied:
Thank you. 

I leave this unchanged then.

javiereguiluz commented on messenger.rst:
```diff
@@ -1268,6 +1268,72 @@ during a request::
+
+.. note::
+
+    The ``wait_time`` parameter define the maximum duration Amazon SQS should
```
`define` -> `defines`

Nyholm replied:
Thanks

javiereguiluz commented on messenger.rst:
```diff
@@ -1268,6 +1268,72 @@ during a request::
+    It helps reducing the cost of using Amazon SQS by eliminating the number
+    of empty responses.
+
+    The ``poll_timeout`` parameter define the duration the receiver should wait
```
`define` -> `defines`

Nyholm replied:
Thanks

javiereguiluz commented on messenger.rst:
```diff
@@ -1268,6 +1268,72 @@ during a request::
+
+.. note::
+
+    If the queue name is suffixed by ``.fifo``, AWS will creates a `FIFO queue`_.
```
`creates` -> `create`

Nyholm replied:
Thank you

javiereguiluz reviewed (approved):
Thanks for contributing these docs!

Nyholm committed:
Typos

OskarStark commented on messenger.rst:
```diff
@@ -1268,6 +1268,72 @@ during a request::
+==================  ======================================  ======================
+     Option         Description                             Default
+==================  ======================================  ======================
+endpoint            Absolute URL to the SQS service         https://sqs.eu-west-1.amazonaws.com
```
Should we sort the option alphabetically?

Nyholm replied:
Sure, doing it now

OskarStark commented:
Note for the mergers, this must target 5.1 branch

Nyholm committed:
Sort alphabetically

Nyholm commented:
I've updated the PR

Nyholm committed:
Remove Michelle from the CARE team

wouterj committed:
minor #14403 Remove Michelle from the CARE team (Nyholm)

This PR was submitted for the 5.x branch but it was merged into the 4.4 branch instead.

Discussion
----------

Remove Michelle from the CARE team

She has resigned.

Commits
-------

cb0c5260c Remove Michelle from the CARE team

wouterj committed:
Merge branch '4.4' into 5.1

* 4.4:
  Remove Michelle from the CARE team
  Update user_provider.rst

wouterj commented:
Thanks @Nyholm and @Simperfit!

During the merge, I mistakingly added some commits that should also have made it in 5.1 (from 4.4). So the diff of this PR is a bit weird, but the result is correct.

Afterwards, I've fixed the comment of @javiereguiluz along with some minor formatting changes in 23fd9b401c6b6014551e7ce423d694809b456ecd

Nyholm commented:
Thank you
